### PR TITLE
Extend sf registry

### PR DESCRIPTION
**Pull request by fatimayousuf**

### CODE REVIEW

File: test/test_fsw_state_field_registry_mock/test_fsw_state_field_registry_mock.cpp [hunk 1]
```diff
@@ -149,6 +149,17 @@ void test_create_writable_vector_field_args() {
     TEST_ASSERT_NOT_NULL(registry.find_writable_field("foo2"));
 }
 
+void test_create_fault() {
+    StateFieldRegistryMock registry;
+
+    registry.create_fault("foo", 1, 3);
+    TEST_ASSERT_NOT_NULL(registry.find_fault("foo"));
```
Comment (tanishqaggarwal) [inline test/test_fsw_state_field_registry_mock/test_fsw_state_field_registry_mock.cpp]: Let's also check that "foo" exists as a writable field?

File: test/test_fsw_state_field_registry_mock/test_fsw_state_field_registry_mock.cpp [hunk 2]
```diff
@@ -164,11 +175,15 @@ void test_clear() {
     registry.create_internal_field<unsigned int>("foo3");
     TEST_ASSERT_NOT_NULL(registry.find_internal_field("foo3"));
 
+    registry.create_fault("foo4", 1, 300);
+    TEST_ASSERT_NOT_NULL(registry.find_fault("foo4"));
+
     registry.clear();
     TEST_ASSERT_NULL(registry.find_internal_field("foo"));
```
Comment (tanishqaggarwal) [inline test/test_fsw_state_field_registry_mock/test_fsw_state_field_registry_mock.cpp]: This as well should check that foo is writable, not internal.

File: test/test_fsw_state_field_registry_mock/test_fsw_state_field_registry_mock.cpp (hunk 1 above)
Comment (tanishqaggarwal) [reply test/test_fsw_state_field_registry_mock/test_fsw_state_field_registry_mock.cpp]: Note that Fault derives from WritableStateField<bool> so foo should be a writable field.

File: test/StateFieldRegistryMock.hpp [hunk 3]
```diff
@@ -292,26 +301,43 @@ class StateFieldRegistryMock : public StateFieldRegistry {
         return field_ptr;
     }
 
+    /**
+     * @brief Create a fault and add it to the registry.
+     * 
+     * @param name Name of fault to create.
+     * @return Pointer to fault that was created.
+     */
+    std::shared_ptr<Fault> create_fault(const std::string& name, const size_t _persistance, unsigned int control_cycle_count)
```
Comment (tanishqaggarwal) [inline test/StateFieldRegistryMock.hpp]: Persistence is spelled with an e not an a

File: test/test_fsw_state_field_registry_mock/test_fsw_state_field_registry_mock.cpp [hunk 4]
```diff
@@ -149,6 +149,17 @@ void test_create_writable_vector_field_args() {
     TEST_ASSERT_NOT_NULL(registry.find_writable_field("foo2"));
```
Comment (tanishqaggarwal) [inline test/test_fsw_state_field_registry_mock/test_fsw_state_field_registry_mock.cpp]: Can you also add tests to the test_common_state_field_registry? It's important that the base-level functionality of adding and finding faults (i.e. without StateFieldRegistryMock) is tested, or flight software may break.

Comment (tanishqaggarwal) [reply test/test_fsw_state_field_registry_mock/test_fsw_state_field_registry_mock.cpp]: I think there were some segfaults when I tried doing it. Have fun 🙃

File: test/StateFieldRegistryMock.hpp (hunk 3 above)
Comment (fatimayousuf) [reply test/StateFieldRegistryMock.hpp]: How very...meticulous

File: test/test_fsw_state_field_registry_mock/test_fsw_state_field_registry_mock.cpp (hunk 4 above)
Comment (fatimayousuf) [reply test/test_fsw_state_field_registry_mock/test_fsw_state_field_registry_mock.cpp]: image

File: test/test_fsw_state_field_registry_mock/test_fsw_state_field_registry_mock.cpp (hunk 2 above)
Comment (fatimayousuf) [reply test/test_fsw_state_field_registry_mock/test_fsw_state_field_registry_mock.cpp]: I'll change it, but I think you wrote that, not me lol

**Pull request merged**